From cc55c04053945bc99309af85b8d81097f1fea927 Mon Sep 17 00:00:00 2001
From: aabacchus <ben@bvnf.space>
Date: Mon, 19 Jul 2021 21:16:40 +0100
Subject: [PATCH] kiss: continue build

---
 kiss | 52 +++++++++++++++++++++++++++++++++-------------------
 1 file changed, 33 insertions(+), 19 deletions(-)

diff --git a/kiss b/kiss
index 9274ff6..d71de40 100755
--- a/kiss
+++ b/kiss
@@ -936,14 +936,7 @@ pkg_build_all() {
         ! [ -f "$repo_dir/sources" ] || pkg_extract  "$pkg"
 
         pkg_build    "$pkg"
-        pkg_manifest "$pkg" "$pkg_dir"
-        pkg_strip    "$pkg"
-
-        cd "$pkg_dir/$pkg/$pkg_db/$pkg"
-
-        pkg_fix_deps "$pkg"
-        pkg_etcsums
-        pkg_tar      "$pkg"
+        pkg_finish_b "$pkg"
 
         if ok "$pkg_update" || ! contains "$explicit" "$pkg"; then
             log "$pkg" "Needed as a dependency or has an update, installing"
@@ -961,6 +954,31 @@ pkg_build_all() {
     esac
 }
 
+pkg_finish_b() {
+    # Finish building a package and create a tarball.
+    pkg="$1"
+    pkg_find "$1"
+
+    # Copy the repository files to the package directory.
+    cp -LRf "$repo_dir" "$pkg_dir/$1/$pkg_db/"
+
+    log "$1" "Successfully built package"
+
+    # arg1: post-build
+    # arg2: package name
+    # arg3: path to DESTDIR
+    run_hook post-build "$1" "$pkg_dir/$1"
+
+    pkg_manifest "$pkg" "$pkg_dir"
+    pkg_strip    "$pkg"
+
+    cd "$pkg_dir/$pkg/$pkg_db/$pkg"
+
+    pkg_fix_deps "$pkg"
+    pkg_etcsums
+    pkg_tar      "$pkg"
+}
+
 pkg_build() {
     # Install built packages to a directory under the package name to
     # avoid collisions with other packages.
@@ -1015,16 +1033,6 @@ pkg_build() {
     # Delete the log file if the build succeeded to prevent the directory
     # from filling very quickly with useless logs.
     equ "$KISS_KEEPLOG" 1 || rm -f "$log_dir/$1-$time-$KISS_PID"
-
-    # Copy the repository files to the package directory.
-    cp -LRf "$repo_dir" "$pkg_dir/$1/$pkg_db/"
-
-    log "$1" "Successfully built package"
-
-    # arg1: post-build
-    # arg2: package name
-    # arg3: path to DESTDIR
-    run_hook post-build "$1" "$pkg_dir/$1"
 }
 
 pkg_checksum() {
@@ -1842,6 +1850,11 @@ args() {
         a|alternatives) pkg_alternatives "$@" ;;
         b|build)        pkg_build_all "${@:?No packages installed}" ;;
         c|checksum)     for pkg do pkg_checksum "$pkg"; done ;;
+        co|continue)
+            export pkg_dir=${PWD%/*}
+            [ "$1" ] || set -- "${PWD##*/}"
+            pkg_finish_b "$1"
+        ;;
         d|download)     for pkg do pkg_source "$pkg"; done ;;
         H|help-ext)     pkg_help_ext "$@" ;;
         i|install)      for pkg do pkg_install "$pkg"; done ;;
@@ -1853,10 +1866,11 @@ args() {
         v|version)      printf '5.5.25\n' ;;
 
         '')
-            log 'kiss [a|b|c|d|i|l|r|s|u|U|v] [pkg]...'
+            log 'kiss [a|b|c|co|d|i|l|r|s|u|U|v] [pkg]...'
             log 'alternatives List and swap alternatives'
             log 'build        Build packages'
             log 'checksum     Generate checksums'
+            log 'continue     Continue a package build in the current dir'
             log 'download     Download sources'
             log 'install      Install packages'
             log 'list         List installed packages'
-- 
2.32.0

